src/main/java/hudson/remoting/Channel.java

```diff
@@ -1080,18 +1079,17 @@ public void terminate(@Nonnull IOException e) {
                     logger.log(Level.WARNING, "Failed to close down the reader side of the transport", x);
                 }
                 try {
-                    synchronized (pendingCalls) {
-                        for (Request<?, ?> req : pendingCalls.values())
-                            req.abort(e);
-                        pendingCalls.clear();
+                    for (Request<?, ?> req : pendingCalls.values()) {
+                        req.abort(e);
                     }
-                    synchronized (executingCalls) {
-                        for (Request<?, ?> r : executingCalls.values()) {
-                            java.util.concurrent.Future<?> f = r.future;
-                            if (f != null) f.cancel(true);
+                    pendingCalls.clear();
```

Comment:
Similar comments here as for pendingCalls. Mixing broken, inconsistent threading models doesn't help.